docs/cmd/graph.md

```diff
@@ -0,0 +1,94 @@
+# graph command
+
+### Description
+
+The `graph` command generates graphical representaitons of the deployed topology.
+
+Two graph styles are available, using [Graphviz](https://graphviz.org/) or using a HTML page generated by `containerlab` based on a Go html template.
+
+#### Graphviz
+
+PLACEHOLDER
+
+#### HTML
+
+The HTML based graph is created by rendering a [GoHTML](https://golang.org/pkg/html/template/) template against a `struct` containing the topology name as well as a `json` string where 2 lists are present: `nodes` and `links` .
+
+`nodes` contains data about the lab nodes, such as name, kind, type, image, state, IP addresses,...
+
+`links` contains the list of links defined by source node and target node, as well as the endpoint names
+
+```json
+{
+  "nodes": [
+    {
+      "name": "node1-1",
+      "image": "srlinux:20.6.1-286",
+      "kind": "srl",
+      "group": "tier-1",
+      "state": "running/Up 21 seconds",
+      "ipv4_address": "172.23.23.3/24",
+      "ipv6_address": "2001:172:23:23::3/80"
+    },
+    // omitted rest of nodes
+  ],
+  "links": [
+    {
+      "source": "node1-2",
+      "source_endpoint": "e1-1",
+      "target": "node2-1",
+      "target_endpoint": "e1-2"
+    },
+    {
+      "source": "node2-1",
+      "source_endpoint": "e1-4",
+      "target": "node3-1",
+      "target_endpoint": "e1-1"
+    },
+    // omitted rest of links 
+  ]
+}
+```
+
+Within the template Javascript libraries such as  [**d3js directed force graph**](https://observablehq.com/collection/@d3/d3-force) or [**vis.js network**](https://visjs.github.io/vis-network/docs/network/) can be used to generate custom topology graphs.
+
+`containerlab` comes with a (minimalistic) default template using [d3js](https://github.com/d3/d3-force).
+
+After the graph generation, it allows to freely move the nodes to a desired position then export the graph in PNG format
+
+![default_graph](../images/html_topo.png)
```

Comment:
I am personally not a fan of keeping binary data within the code repo.

Although I do not expect this repo to have lots of images, maybe its ok to use the cloud storage and load it from the url?

working substitution:
`![default_graph](https://gitlab.com/rdodin/pics/-/wikis/uploads/5f3ade3559a5f044d4786bfd0e278b65/image.png`